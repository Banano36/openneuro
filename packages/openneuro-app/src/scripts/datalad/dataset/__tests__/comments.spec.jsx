import React from 'react'
import { mount } from 'enzyme'
import Comments from '../comments.jsx'

const emptyState =
  '{"blocks":[{"key":"3sm42","text":"","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}}],"entityMap":{}}'
const exampleComment = {
  id: 'xyz',
  text: emptyState,
  user: { id: '1234', email: 'example@example.com' },
  replies: [],
}

describe('Comments component', () => {
  it('renders one top level comment', () => {
    const wrapper = mount(
      <Comments datasetId="ds000001" comments={[exampleComment]} />,
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('recursively renders a tree of comments', () => {
    const comments = [{ ...exampleComment, replies: [exampleComment] }]
    const wrapper = mount(
      <Comments datasetId="ds000001" comments={[exampleComment]} />,
    )
  })
})
