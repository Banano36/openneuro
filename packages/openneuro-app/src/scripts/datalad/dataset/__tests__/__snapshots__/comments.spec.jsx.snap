// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments component recursively renders a tree of comments 1`] = `
<Comments
  comments={
    Array [
      Object {
        "id": "xyz",
        "replies": Array [
          Object {
            "id": "xyz",
            "replies": Array [],
            "text": "{\\"blocks\\":[{\\"key\\":\\"3sm42\\",\\"text\\":\\"\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[],\\"entityRanges\\":[],\\"data\\":{}}],\\"entityMap\\":{}}",
            "user": Object {
              "email": "example@example.com",
              "id": "1234",
            },
          },
        ],
        "text": "{\\"blocks\\":[{\\"key\\":\\"3sm42\\",\\"text\\":\\"\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[],\\"entityRanges\\":[],\\"data\\":{}}],\\"entityMap\\":{}}",
        "user": Object {
          "email": "example@example.com",
          "id": "1234",
        },
      },
    ]
  }
  datasetId="ds000001"
>
  <div
    className="col-xs-12"
  >
    <div
      className="dataset-comments"
    >
      <h2>
        Comments
      </h2>
      <ErrorBoundary
        subject="error in dataset comments"
      >
        <LoggedIn />
        <LoggedOut>
          <div>
            Please sign in to contribute to the discussion.
          </div>
        </LoggedOut>
        <CommentTree
          commentMap={
            Object {
              "xyz": Object {
                "id": "xyz",
                "replies": Array [
                  Object {
                    "id": "xyz",
                    "replies": Array [],
                    "text": "{\\"blocks\\":[{\\"key\\":\\"3sm42\\",\\"text\\":\\"\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[],\\"entityRanges\\":[],\\"data\\":{}}],\\"entityMap\\":{}}",
                    "user": Object {
                      "email": "example@example.com",
                      "id": "1234",
                    },
                  },
                ],
                "text": "{\\"blocks\\":[{\\"key\\":\\"3sm42\\",\\"text\\":\\"\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[],\\"entityRanges\\":[],\\"data\\":{}}],\\"entityMap\\":{}}",
                "user": Object {
                  "email": "example@example.com",
                  "id": "1234",
                },
              },
            }
          }
          comments={Array []}
          datasetId="ds000001"
        />
      </ErrorBoundary>
    </div>
  </div>
</Comments>
`;

exports[`Comments component renders one top level comment 1`] = `
<Comments
  comments={
    Array [
      Object {
        "id": "xyz",
        "replies": Array [],
        "text": "{\\"blocks\\":[{\\"key\\":\\"3sm42\\",\\"text\\":\\"\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[],\\"entityRanges\\":[],\\"data\\":{}}],\\"entityMap\\":{}}",
        "user": Object {
          "email": "example@example.com",
          "id": "1234",
        },
      },
    ]
  }
  datasetId="ds000001"
>
  <div
    className="col-xs-12"
  >
    <div
      className="dataset-comments"
    >
      <h2>
        Comments
      </h2>
      <ErrorBoundary
        subject="error in dataset comments"
      >
        <LoggedIn />
        <LoggedOut>
          <div>
            Please sign in to contribute to the discussion.
          </div>
        </LoggedOut>
        <CommentTree
          commentMap={
            Object {
              "xyz": Object {
                "id": "xyz",
                "replies": Array [],
                "text": "{\\"blocks\\":[{\\"key\\":\\"3sm42\\",\\"text\\":\\"\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[],\\"entityRanges\\":[],\\"data\\":{}}],\\"entityMap\\":{}}",
                "user": Object {
                  "email": "example@example.com",
                  "id": "1234",
                },
              },
            }
          }
          comments={Array []}
          datasetId="ds000001"
        />
      </ErrorBoundary>
    </div>
  </div>
</Comments>
`;
